src/main/java/com/gmail/woosay333/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -62,4 +65,12 @@ public List<BookDto> search(BookSearchParameters searchParameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    private void bookIsbnAlreadyExistsCheck(Book book) {
+        if (book != null) {
+            throw new BookIsbnAlreadyExistsException(
+                    String.format("Book with isbn: %s already exists", book.getIsbn())
+            );
+        }
```

Comment:
Better to call a repository in this method, get an Optional, and use `orElseThrow()`, or a `repo.existsByIsbn(isbn)` which will return a boolean - will probably be faster because we do not actually need to fetch the data that we are looking for